src/cpp/session/modules/SessionFind.cpp

```diff
@@ -629,87 +630,108 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
-                      matchSize != replaceString.size())
+                      matchSize != replaceSize)
                   {
                      json::Array tempMatchOn(*pReplaceMatchOn);
                      json::Array tempMatchOff(*pReplaceMatchOff);
                      pReplaceMatchOn->clear();
                      pReplaceMatchOff->clear();
 
-                     // put back in reverse order
-                     int offset(gsl::narrow_cast<int>(replaceString.size() - matchSize));
+                     // put back in reverse order for the frontend
+                     int offset(gsl::narrow_cast<int>(replaceSize - matchSize));
                      pReplaceMatchOn->push_back(json::Value(gsl::narrow_cast<int>(matchOn)));
                      pReplaceMatchOff->push_back(json::Value(gsl::narrow_cast<int>(replaceMatchOff)));
-                     addOffsetIntegerToJsonArray(tempMatchOn, offset, pReplaceMatchOn);
-                     addOffsetIntegerToJsonArray(tempMatchOff, offset, pReplaceMatchOff);
+                     substractOffsetIntegersToJsonArray(tempMatchOn, offset, pReplaceMatchOn);
+                     substractOffsetIntegersToJsonArray(tempMatchOff, offset, pReplaceMatchOff);
                   }
-                  else if (lineSuccess)
+                  else
                   {
                      pReplaceMatchOn->push_back(json::Value(gsl::narrow_cast<int>(matchOn)));
                      pReplaceMatchOff->push_back(json::Value(gsl::narrow_cast<int>(replaceMatchOff)));
                   }
                }
-               if (!findResults().preview())
-                  pProgress->addUnit();
                pos--;
             }
+            // encode and write the new line to file
             if (!findResults().preview())
             {
                Error error = encodeAndWriteLineToFile(pLineInfo->decodedContents,
-                  pLineInfo->leftContents, pLineInfo->rightContents);
+                  pLineInfo->leadingWhitespace, pLineInfo->trailingWhitespace);
                if (error)
-                  addErrorMessage(error.asString(), pErrorMessage,
-                     pReplaceMatchOn, pReplaceMatchOff, &lineSuccess);
+                  addReplaceErrorMessage(error.asString(), pErrorMessage, pReplaceMatchOn,
+                     pReplaceMatchOff, &lineSuccess);
+
             }
          }
       }
       return Success();
    }
 
+   bool shouldSkipFile(std::string file)
+   {
+      bool skipFile = false;
+      if (file.find("/.Rproj.user/") != std::string::npos ||
+          file.find("/.git/") != std::string::npos ||
+          file.find("/.svn/") != std::string::npos ||
+          file.find("/packrat/lib/") != std::string::npos ||
+          file.find("/packrat/src/") != std::string::npos ||
+          file.find("/renv/library/") != std::string::npos ||
+          file.find("/.Rhistory") != std::string::npos)
+         skipFile = true;
+      return (skipFile);
```

Comment:
I think you should be safe to just return the if condition (i.e. `return (file.find("...") || ... );

Reply:
Done!